### PR TITLE
Set up GHA CI

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Someone with admin privileges on this repo needs to create a new Documenter key, and add the private key (Base64-encoced) as a GitHub secret named `DOCUMENTER_KEY`: at the top of this page, click on Settings, then click on Secrets in the left-hand sidebar.

And then add the corresponding public key as a deploy key with write access on this repo.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.0'
+          - '1' automatically expands to the latest stable 1.x release of Julia.
```
Comment (christopher-dG) [inline .github/workflows/ci.yml:19]:
```suggestion
          - '1'  # automatically expands to the latest stable 1.x release of Julia.
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        version:
+          - '1.0'
+          - '1'  # automatically expands to the latest stable 1.x release of Julia.
+          - 'nightly'
+        os:
+          - ubuntu-latest
```
Comment (KristofferC) [inline .github/workflows/ci.yml:22]: missing mac and windows?

Comment (DilumAluthge) [reply .github/workflows/ci.yml:22]: Done

**Comment by DilumAluthge** [comment]
Good to go now? @KristofferC @christopher-dG

**Comment by DilumAluthge** [comment]
The Windows failure on Julia nightly is https://github.com/JuliaLang/julia/issues/38409

**Pull request merged**